File: app/src/main/java/me/chrislane/accudrop/db/Position.java
package me.chrislane.accudrop.db;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.ForeignKey;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import java.util.Date;

@Entity(
        tableName = "position",
        indices = {
                @Index("jump_id")
        },
        foreignKeys = {
                @ForeignKey(entity = Jump.class,
                        parentColumns = "id",
                        childColumns = "jump_id")
        }
)
public class Position {
    @PrimaryKey(autoGenerate = true)
    public int id;

    @NonNull
    public String useruuid;

    public Integer altitude;

    public Double vspeed;

    public Float hspeed;

    public Double latitude;

    public Double longitude;

    public Date time;

    public FallType fallType;

    @ColumnInfo(name = "jump_id")
    @NonNull
    public int jumpId;
}

File: app/src/main/java/me/chrislane/accudrop/db/Position.kt
package me.chrislane.accudrop.db

import androidx.room.*
import java.util.*

@Entity(
    tableName = "position",
    indices = [Index("jump_id")],
    foreignKeys = [
      ForeignKey(entity = Jump::class, parentColumns = ["id"], childColumns = ["jump_id"]),
      ForeignKey(entity = User::class, parentColumns = ["uuid"], childColumns = ["user_id"])])
data class Position(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    @ColumnInfo(name = "jump_id")
    val jumpId: Int,
    @ColumnInfo(name = "user_id")
    val userUuid: UUID,
    val time: Date,
    val altitude: Int?,
    @ColumnInfo(name = "vertical_speed")
    val vSpeed: Double?,
    @ColumnInfo(name = "horizontal_speed")
    val hSpeed: Float?,
    val latitude: Double?,
    val longitude: Double?,
    @ColumnInfo(name = "fall_type")
    val fallType: FallType)